### PR TITLE
[DataObject] Fix block inside localized field

### LINKED ISSUE #6976: block inside localized field
# Bug Report
I am currently having troubles when I try to use a block inside a localized field. You can see the class structure in the attached 
image 
Class

### Actual behavior
If I am creating a new blockitem inside the localized field and save it everything seems to work but as soon as I reload pimcore and try to save the Object again the backend instantly goes into maintenance mode.

### Steps to reproduce
create class -> create object -> add block item -> save -> reload pimcore -> try to save again

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #6976

### CODE REVIEW

**Review by weisswurstkanone — APPROVED** [review]

**Review by weisswurstkanone — CHANGES_REQUESTED** [review]
unit tests

**Comment by dvesh3** [comment]
> unit tests

@weisswurstkanone reverted

**Comment by weisswurstkanone** [comment]
@dvesh3  still failing

File: models/DataObject/Data/BlockElement.php [hunk 1]
```diff
@@ -150,7 +148,17 @@ function ($currentValue) {
...
+                /**
+                 * @param mixed $element
+                 *
+                 * @return bool
+                 */
+                public function matches($element)
+                {
+                    return $element instanceof ElementInterface;
+                }
+            }
```
Comment (jdreesen) [inline models/DataObject/Data/BlockElement.php:161]: Why do you need an anonymous class here? Shouldn't this be sufficient?

```suggestion
            new TypeMatcher(ElementInterface::class)
```

**Comment by weisswurstkanone** [comment]
We will come up with a generic approach.

Discussion points:
* https://github.com/pimcore/pimcore/pull/5799
* https://github.com/pimcore/pimcore/pull/6236

**Pull request closed without merging**